packages/commonwealth/server/passport/magic.ts

```diff
@@ -454,11 +458,21 @@ async function magicLoginRoute(
         throw new Error('sessionPayload signed with invalid signature');
       }
     }
+
     if (chainToJoin) {
-      if (
-        chainToJoin.base === ChainBase.CosmosSDK &&
-        session.payload.chain.startsWith('cosmos:')
-      ) {
+      if (chainToJoin.base === ChainBase.CosmosSDK) {
+        const signaturePattern = /(.+)\/([A-Za-z0-9+]+)\/([A-Za-z0-9+=]+)/;
+        const signaturePatternMatch = signaturePattern.exec(session.signature);
+        if (signaturePatternMatch === null) {
+          throw new Error(
+            `Invalid signature: signature did not match ${signaturePattern}`,
+          );
+        }
+        const [_, domain, nonce, signatureData] = signaturePatternMatch;
+
+        // throws if magicAddress does not match signed address in session DIDToken
+        await magic.token.validate(signatureData, req.body.magicAddress);
```

Comment:
might not need to do it here since we validate to get the session signature? 

It also can be reverified if session keys enforced.